### PR TITLE
docs: add feature issue template for adding contributors

### LINKED ISSUE #203: docs: more adjustments to the PR template
### Type of feature

📝 Documentation

### Current behavior

When filling out the form for docs-related PRs, I found myself struggling with answering some of the questions and connecting the corresponding issue to it. A similar phenomenon could happen for people who want to do translation contributions or improve the directions in the README and Contributing files.  

### Suggested solution

1. Put a `[] Not applicable` checkbox in the [Contributors Checklist section](https://github.com/open-sauced/guestbook/blob/main/.github/PULL_REQUEST_TEMPLATE.md#contributors-checklist-check-all-applicable)
2. Put a `[] 📝 CONTRIBUTING.md` checkbox in the [Added to Documentation section](https://github.com/open-sauced/guestbook/blob/main/.github/PULL_REQUEST_TEMPLATE.md#added-to-documentation). 
3. Put a `## Closes Issue` section after the [What Type of PR is this? section](https://github.com/open-sauced/guestbook/blob/main/.github/PULL_REQUEST_TEMPLATE.md#what-type-of-pr-is-this-check-all-applicable)

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### LINKED ISSUE #206: Docs: Create an issue template to add users as contributors
## Description

To get our "Intro to Open Source" course participants into the flow of contributing to open source, we want them to learn  and experience creating an issue and link their PR to it later on.

Originally, we will let contributors to create an issue from scratch. However, adding an issue template would be a good practice for them to learn about issue's structure.

**Based on these discussions**:
- _Originally posted by @adiati98 in https://github.com/open-sauced/guestbook/issues/203#issuecomment-1875519001_
- _Originally posted by @BekahHW in https://github.com/open-sauced/guestbook/issues/203#issuecomment-1875991234_

## Suggested Solution

To make it easier for contributors and maintainers to triage, let's create a simple issue template with below sections and add instructions (coment them out) to each section:

- [x] 1. Create an `ISSUE_TEMPLATE` folder inside `.github` folder.
- [x] 2. Create a file call `feature_request.yml`.
- [x] 3. Add these to the template (*See our [Feature Request template](https://github.com/open-sauced/.github/blob/main/.github/ISSUE_TEMPLATE/feature_request.yml?plain=1) to help you out*):
    - [ ] name: 🚀 Feature request
    - [ ] description: Add contributors to the guest book 💡 
    - [ ] title: "Feature: "
    - [ ] labels: [💡 feature]
    - [ ] All have `validations: required: true`

    Atrributes' label in the body:
    - [ ] Description 
           ***description**: Please describe the issue — for example, "As part of the Intro to Open Source course, we are encouraged to contribute to this repo."*
    - [ ] Suggested solution
          ***description**: Suggest a solution — for example, "Add myself as a contributor."*
    - [ ] Code of Conduct agreement
    - [ ] Getting Started Instructions agreement

        ```yml
        type: checkboxes
        id: contribution
        attributes:
          label: Getting Started Instructions
          description: Please read [Getting Started](https://intro.opensauced.pizza/#/05-how-to-contribute-to-open-source?id=getting-started) to add yourself as a contributor.
          options:
            - label: I agree to follow this project's Getting Started instructions.
              required: true
        ```

### PR DESCRIPTION
**Pull request by CBID2**
## Description
This PR adds a feature issue template for adding contributors. This will encourage people who have taken the Intro to Open Source course to practice raising and writing an effective issue. 

## Related Tickets & Documents 
Closes #206 
Closes #203 

## What type of PR is this? (check all applicable)

- [ ] 🤝 Add a contributor
- [ ] 📝 Documentation Update
- [x] 🍕 Feature Request

### I've read through the [Getting Started](https://intro.opensauced.pizza/#/05-how-to-contribute-to-open-source?id=getting-started) section.

- [ ] ✅ Yes
- [ ] ❌ Not yet
- [x] 🙅🏾 Not Applicable 

### How did you add yourself as a contributor?

- [ ] 🤖 With CLI
- [ ] ⌨️ Manually
- [ ] 🙅🏾 Not Applicable 
#### If you added yourself manually, did you follow the [emoji key and contribution types](https://allcontributors.org/docs/en/emoji-key) to fill in the value?

- [ ] ✅ Yes
- [ ] ❌ No
- [x] 🙅🏾 Not applicable

### Have you run `npm run contributors:generate` to generate your profile and the badge on the README?
- [ ] ✅ Yes
- [ ] ❌ No
- [x] 🙅🏾 Not Applicable 
## Added to documentation?

- [ ] 📜 README.md
- [x] 🙅 no documentation needed

## [optional] What GIF best describes this PR or how it makes you feel?
alfred

### CODE REVIEW

**Comment by CBID2** [comment]
Hey @adiati98. I made the PR. It's in draft mode because I stumbled on [Step 3](https://github.com/open-sauced/guestbook/issues/206). Can you check to see if the format for this part is correct?

**Comment by adiati98** [comment]
@CBID2 we want to customize this template for guestbook repo.
So, we will only use the attributes that are mentioned in the issue. 👍

**Comment by CBID2** [comment]
> @CBID2 we want to customize this template for guestbook repo.
> 
> So, we will only use the attributes that are mentioned in the issue. 👍

@adiati98. I get but I got stuck with formatting certain parts

**Comment by adiati98** [comment]
@CBID2 the format of step 3 looks good.

**Comment by CBID2** [comment]
> @CBID2 the format of step 3 looks good.

Ok good so anything I should change @adiati98?

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        - "🎨 Style"
+        - "🧑‍💻 Refactor"
+        - "🔥 Performance"
+        - "✅ Test"
+        - "🤖 Build"
+        - "🔁 CI"
+        - "📦 Chore"
+        - "⏩ Revert"
+    validations:
+      required: true
```
Comment (adiati98) [inline .github/workflows/ISSUE_TEMPLATE/feature_request.yml:28]: This part is not mentioned in the issue.

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  - type: textarea
+    attributes:
+      label: Current behavior
```
Comment (adiati98) [inline .github/workflows/ISSUE_TEMPLATE/feature_request.yml:31]:
```suggestion
      label: Description
```

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      required: true
+  - type: textarea
+    attributes:
+      label: Suggested solution
+      description: Suggest a solution — for example, "Add myself as a contributor."
+  - type: input
+    id: context
+    attributes:
+      label: Additional context
+      description: Add any other context about the problem or helpful links here.
```
Comment (adiati98) [inline .github/workflows/ISSUE_TEMPLATE/feature_request.yml:43]: This part is also not mentioned in the issue.

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      label: Code of Conduct
+      description: By submitting this issue, you agree to follow our [Code of Conduct](https://docs.opensauced.pizza/contributing/code-of-conduct/)
+      options:
+        - label: I agree to follow this project's Code of Conduct
+          required: true
+  - type: checkboxes
+    id: contribution
+    attributes:
+      label: Getting Started Instructions
+      description: Please read[Getting Started](https://intro.opensauced.pizza/#/05-how-to-contribute-to-open-source?id=getting-started) to add yourself as a contributor.
```
Comment (adiati98) [inline .github/workflows/ISSUE_TEMPLATE/feature_request.yml:56]:
```suggestion
      description: Please read [Getting Started](https://intro.opensauced.pizza/#/05-how-to-contribute-to-open-source?id=getting-started) to add yourself as a contributor.
```

**Review by adiati98 — CHANGES_REQUESTED** [review]
@CBID2 I left some reviews here. :)
Can you please also double check with issue? Thanks!

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml (hunk 3 above)
Comment (CBID2) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml:43]: Yeah I'm still figuring out how to fix it

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml (hunk 1 above)
Comment (CBID2) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml:28]: Still trying to figure out what to remove

Comment (adiati98) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml:28]: The whole type. :)

You can test by creating a repo and copy paste what you have here in that repo.
Then further test it by creating an issue.

Comment (CBID2) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml:28]: Ooh

**Comment by CBID2** [comment]
Hi @adiati98! :) I think I made the changes! :)

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          required: true
+  - type: checkboxes
+    id: contribution
+    attributes:
+      label: Getting Started Instructions
+      description: Please read [Getting Started](https://intro.opensauced.pizza/#/05-how-to-contribute-to-open-source?id=getting-started) to add yourself as a contributor.
+      options:
+        - label: I agree to follow this project's Getting Started instructions
+          required: true
+      description: Please read [Getting Started](https://intro.opensauced.pizza/#/05-how-to-contribute-to-open-source?id=getting-started) to add yourself as a contributor.
```
Comment (BekahHW) [inline .github/workflows/ISSUE_TEMPLATE/feature_request.yml:37]: I'm not sure why this description is repeated.

**Review by BekahHW — CHANGES_REQUESTED** [review]

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml (hunk 5 above)
Comment (CBID2) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml:37]: > I'm not sure why this description is repeated. 

Oops that was a mistake @BekahHW

Comment (CBID2) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml]: Made the change @BekahHW

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml (hunk 1 above)
Comment (adiati98) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml]: > Still trying to figure out what to remove

It'll be easier for you to fix format or remove parts when you can see the form live.

Thanks for removing! 👍

File: .github/workflows/ISSUE_TEMPLATE/feature_request.yml (hunk 3 above)
Comment (adiati98) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml:20]: Unless you have another thought here, let's remove this part and we're good to go. 👍

Comment (CBID2) [reply .github/workflows/ISSUE_TEMPLATE/feature_request.yml]: Done @adiati98

**Review by adiati98 — APPROVED** [review]
go time gif

**Review by BekahHW — APPROVED** [review]

**Pull request merged**